File: scripts/dca.py
# -*- coding: utf-8 -*-
import pandas as pd
import cornpy as cp
import matplotlib.pyplot as plt
from adjustText import adjust_text

to_print = True

# Leitura dos dados
matriz = pd.read_csv("especies.csv", header=0, index_col=0, encoding="utf8")

# Transposição da matriz de dados
matriz = matriz.transpose()

# Análise de correspondências corrigida
# utilizando a função decorana() do pacote CEPy
site_scores, species_scores = cp.decorana(matriz)

# Diagrama de dispersão
fig, ax = plt.subplots()
x = list(site_scores[:,0])
y = list(site_scores[:,1])
if to_print:
    plt.scatter(x, y, color="black")
else:
    plt.scatter(x, y, color="blue")
texts = []
for i, txt in enumerate(list(matriz.index)):
    texts.append(ax.text(x[i], y[i], txt))
adjust_text(texts)
plt.axhline(y=0, color='gray', linestyle='--')
plt.axvline(x=0, color='gray', linestyle='--')
plt.xlabel('DCA1')
plt.ylabel('DCA2')
plt.tight_layout()
plt.show()

File: scripts/r/dca.R
to_print <- TRUE

# Leitura dos dados
matriz <- read.csv("especies.csv", row.names=1)

# Transposição da matriz de dados
matriz <- t(matriz)

# Análise de correspondências corrigida
# utilizando a função "decorana" do pacote "vegan"
library(vegan)
dca <- decorana(matriz)
plot(dca, type="none", display="sites", main="")
if (to_print) {
  points(dca, pch=19, cex=1, col="black")
} else {
  points(dca, pch=19, cex=1, col="blue")
}
text(dca, labels=rownames(matriz), cex=1, pos=3)